<a href="https://colab.research.google.com/github/davidsadovy/data690_fall2022/blob/main/assignment_04/assignment_04.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# DATA 690 Assignment 04
- Dave Sadovy
- 9/25/2022

##### 1.  Explore College Scorecard website - https://collegescorecard.ed.gov/

##### 2.  Explore College Scorecard Data web page - https://collegescorecard.ed.gov/data/

##### 3.  Download and read the documentation - https://collegescorecard.ed.gov/assets/FullDataDocumentation.pdf

##### 4.  Download the dataset - https://ed-public-download.app.cloud.gov/downloads/CollegeScorecard_Raw_Data.zip (265MB Zip File)

##### 5.  Unzip the file into a folder in your personal computer (there are many files, one for each school year)

##### 6.  Upload the 2017-2018 school year file "MERGED2017_18_PP.csv" to Colab/Deepnote or any Cloud environment of your choice (you can use your local environment too) and create a Jupyter notebook to explore the dataset



In [17]:
import pandas as pd

# read in source csv file and save as data frame
df = pd.read_csv("/content/MERGED2017_18_PP.csv")       

- How many colleges or rows?
- How many variables or columns?



In [18]:
df.shape    # shows number of rows and columns in data frame

(7112, 1986)

df.shape shows us that the dataset has 7112 colleges or rows, and 1986 variables or columns.

- Look at the first 5 colleges.

In [19]:
df.head()   # shows first 5 rows of data frame

UNITID    OPEID  OPEID6                               INSTNM        CITY  \
0  100654   100200    1002             Alabama A & M University      Normal   
1  100663   105200    1052  University of Alabama at Birmingham  Birmingham   
2  100690  2503400   25034                   Amridge University  Montgomery   
3  100706   105500    1055  University of Alabama in Huntsville  Huntsville   
4  100724   100500    1005             Alabama State University  Montgomery   

  STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
0     AL       35762           NaN      NaN     NaN  ...   
1     AL  35294-0110           NaN      NaN     NaN  ...   
2     AL  36117-3553           NaN      NaN     NaN  ...   
3     AL       35899           NaN      NaN     NaN  ...   
4     AL  36104-0271           NaN      NaN     NaN  ...   

   OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN   G12MN  \
0                           NaN       0.7106        0.7418   5207.0  1185.0   
1                           NaN       0.4009        0.5488  13308.0  9888.0   
2                           NaN       0.5000        0.5000    431.0   437.0   
3                           NaN       0.2012        0.4231   7519.0  2378.0   
4                           NaN       0.7594        0.7402   5229.0   687.0   

   SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
0   1410.0                 NaN                      NaN   
1   1948.0                 NaN                      NaN   
2      4.0                 NaN                      NaN   
3   1203.0                 NaN                      NaN   
4   1143.0                 NaN                      NaN   

   FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
0                       NaN             NaN  
1                       NaN             NaN  
2                       NaN             NaN  
3                       NaN             NaN  
4                       NaN             NaN  

[5 rows x 1986 columns]

- Look at the last 5 colleges.

In [20]:
df.tail()   # shows last 5 rows of data frame

UNITID    OPEID  OPEID6  \
7107  48312404   108163    1081   
7108  48387801  4223701   42237   
7109  48400201  4228101   42281   
7110  48778201  4221501   42215   
7111  49005401  4182601   41826   

                                                 INSTNM             CITY  \
7107                   Arizona State University at Yuma             Yuma   
7108  Bay Area Medical Academy - San Jose Satellite ...         San Jose   
7109           High Desert Medical College - Bakerfield      Bakersfield   
7110                                       BCI - Malden           Malden   
7111               HCI College - Fort Lauderdale Campus  Fort Lauderdale   

     STABBR        ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
7107     AZ  853656900           NaN      NaN     NaN  ...   
7108     CA      95113           NaN      NaN     NaN  ...   
7109     CA      93301           NaN      NaN     NaN  ...   
7110     MA  021480000           NaN      NaN     NaN  ...   
7111     FL      33309           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN  UG12MN  G12MN  \
7107                           NaN          NaN           NaN     NaN    NaN   
7108                           NaN          NaN           NaN     NaN    NaN   
7109                           NaN          NaN           NaN     NaN    NaN   
7110                           NaN          NaN           NaN     NaN    NaN   
7111                           NaN          NaN           NaN     NaN    NaN   

      SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
7107      NaN                 NaN                      NaN   
7108      NaN                 NaN                      NaN   
7109      NaN                 NaN                      NaN   
7110      NaN                 NaN                      NaN   
7111      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
7107                       NaN             NaN  
7108                       NaN             NaN  
7109                       NaN             NaN  
7110                       NaN             NaN  
7111                       NaN             NaN  

[5 rows x 1986 columns]

- Look at a random sample of 5 colleges.

In [21]:
df.sample(5)    # shows 5 random rows of data frame

UNITID    OPEID  OPEID6  \
4065    260691  2217101   22171   
6775  15116602  2158401   21584   
2867    208026  1005303   10053   
5710    459213  4169800   41698   
6270    483814  4221000   42210   

                                               INSTNM          CITY STABBR  \
4065                      Pima Medical Institute-Mesa          Mesa     AZ   
6775                        Harrison College-Columbus      Columbus     IN   
2867       Indian Capital Technology Center-Tahlequah     Tahlequah     OK   
5710                  Gurnick Academy of Medical Arts     San Mateo     CA   
6270  Standard Healthcare Services-College of Nursing  Falls Church     VA   

            ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
4065      85202           NaN      NaN     NaN  ...   
6775  472031843           NaN      NaN     NaN  ...   
2867      74464           NaN      NaN     NaN  ...   
5710      94403           NaN      NaN     NaN  ...   
6270      22043           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN  UG12MN  G12MN  \
4065                           NaN       0.4645        0.6011  1526.0    NaN   
6775                           NaN          NaN           NaN     NaN    NaN   
2867                           NaN       0.5932        0.0000   328.0    NaN   
5710                           NaN       0.6628        0.8779  1770.0    NaN   
6270                           NaN       0.7000        0.9000   390.0    NaN   

      SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
4065   1449.0                 NaN                      NaN   
6775      NaN                 NaN                      NaN   
2867     59.0                 NaN                      NaN   
5710    172.0                 NaN                      NaN   
6270     40.0                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
4065                       NaN             NaN  
6775                       NaN             NaN  
2867                       NaN             NaN  
5710                       NaN             NaN  
6270                       NaN             NaN  

[5 rows x 1986 columns]

##### 7.  Since there are too many variables to explore, many of which are not interesting, create a smaller data frame with the following 9 variables:
- UNITID
- INSTNM
- CITY
- STABBR
- ZIP
- ADM_RATE
- UGDS
- TUITIONFEE_IN
- MN_EARN_WNE_P10

In [23]:
# define new data frame which only includes the 9 selected columns
df2 = df[["UNITID", "INSTNM", "CITY", "STABBR", "ZIP", "ADM_RATE", "UGDS", "TUITIONFEE_IN", "MN_EARN_WNE_P10"]]

# show the first 2 rows to confirm the 9 selected columns
df2.head(2)

UNITID                               INSTNM        CITY STABBR         ZIP  \
0  100654             Alabama A & M University      Normal     AL       35762   
1  100663  University of Alabama at Birmingham  Birmingham     AL  35294-0110   

   ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
0    0.9027   4824.0         9857.0              NaN  
1    0.9181  12866.0         8328.0              NaN

##### 8.  Download and read (use naked eyes, no coding) the data dictionary file - https://collegescorecard.ed.gov/assets/CollegeScorecardDataDictionary.xlsx

##### 9.  Find out the definition of the above 9 variables and document the definitions in the Jupyter Notebook in a Markdown cell so that the readers of your notebook will know the meaning of these variables.

Variable Definitions:
- UNITID - Unit ID for institution.
- INSTNM - Institution name.
- CITY - City.
- STABBR - State postcode.
- ZIP - Zip code.
- ADM_RATE - Admission rate.
- UGDS - Enrollment of undergraduate certificate/degree-seeking students.
- TUITIONFEE_IN - In-state tuition and fees.
- MN_EARN_WNE_P10 - Mean earnings of students working and not enrolled 10 years after entry.

##### 10.  Explore the smaller data frame.

- Look at the first 10 colleges.

In [24]:
df2.head(10)    # shows first 10 rows of smaller data frame

UNITID                               INSTNM            CITY STABBR  \
0  100654             Alabama A & M University          Normal     AL   
1  100663  University of Alabama at Birmingham      Birmingham     AL   
2  100690                   Amridge University      Montgomery     AL   
3  100706  University of Alabama in Huntsville      Huntsville     AL   
4  100724             Alabama State University      Montgomery     AL   
5  100751            The University of Alabama      Tuscaloosa     AL   
6  100760    Central Alabama Community College  Alexander City     AL   
7  100812              Athens State University          Athens     AL   
8  100830      Auburn University at Montgomery      Montgomery     AL   
9  100858                    Auburn University          Auburn     AL   

          ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
0       35762    0.9027   4824.0         9857.0              NaN  
1  35294-0110    0.9181  12866.0         8328.0              NaN  
2  36117-3553       NaN    318.0         6900.0              NaN  
3       35899    0.8123   6917.0        10280.0              NaN  
4  36104-0271    0.9787   4189.0        11068.0              NaN  
5  35487-0100    0.5330  32387.0        10780.0              NaN  
6       35010       NaN   1404.0         4440.0              NaN  
7       35611       NaN   2801.0            NaN              NaN  
8  36117-3596    0.8240   4211.0         8020.0              NaN  
9       36849    0.8393  23391.0        10968.0              NaN

- Look at the last 10 colleges.

In [25]:
df2.tail(10)    # shows last 10 rows of smaller data frame

UNITID                                             INSTNM  \
7102  48154401  National Personal Training Institute of Cleveland   
7103  48305401             Barber School of Pittsburgh - Ambridge   
7104  48312401                 Arizona State University at Tucson   
7105  48312402  Arizona State University - ASU Colleges at Lak...   
7106  48312403        Arizona State University at The Gila Valley   
7107  48312404                   Arizona State University at Yuma   
7108  48387801  Bay Area Medical Academy - San Jose Satellite ...   
7109  48400201           High Desert Medical College - Bakerfield   
7110  48778201                                       BCI - Malden   
7111  49005401               HCI College - Fort Lauderdale Campus   

                  CITY STABBR        ZIP  ADM_RATE  UGDS  TUITIONFEE_IN  \
7102  Highland Heights     OH      44143       NaN   NaN            NaN   
7103          Ambridge     PA  150030000       NaN   NaN            NaN   
7104            Tucson     AZ  857454284       NaN   NaN            NaN   
7105  Lake Havasu City     AZ  864036877       NaN   NaN         6904.0   
7106          Thatcher     AZ  855525545       NaN   NaN            NaN   
7107              Yuma     AZ  853656900       NaN   NaN            NaN   
7108          San Jose     CA      95113       NaN   NaN            NaN   
7109       Bakersfield     CA      93301       NaN   NaN            NaN   
7110            Malden     MA  021480000       NaN   NaN            NaN   
7111   Fort Lauderdale     FL      33309       NaN   NaN        22575.0   

      MN_EARN_WNE_P10  
7102              NaN  
7103              NaN  
7104              NaN  
7105              NaN  
7106              NaN  
7107              NaN  
7108              NaN  
7109              NaN  
7110              NaN  
7111              NaN

- Look at a random sample of 10 colleges.

In [26]:
df2.sample(10)   # shows 10 random rows of smaller data frame

UNITID                                             INSTNM  \
5028  443854                  Altierus Career College-Arlington   
3246  218663              University of South Carolina-Columbia   
3307  219879                  Cleveland State Community College   
5763  460215                                  Ecotech Institute   
3541  228644  The University of Texas Health Science Center ...   
1955  179715                     North Central Missouri College   
2152  187620               New Mexico State University-Dona Ana   
3082  214634  Pennsylvania State University-Penn State Shenango   
5196  447795              American Institute of Massage Therapy   
600   129482                   American Institute-West Hartford   

               CITY STABBR         ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  \
5028      Arlington     TX  76011-6630       NaN    210.0        11274.0   
3246       Columbia     SC       29208    0.7231  25550.0        11862.0   
3307      Cleveland     TN       37312       NaN   2163.0         4139.0   
5763         Aurora     CO       80012       NaN    280.0            NaN   
3541    San Antonio     TX  78229-3900       NaN    807.0            NaN   
1955        Trenton     MO       64683       NaN   1171.0         3510.0   
2152     Las Cruces     NM       88011       NaN   5600.0         1728.0   
3082         Sharon     PA  16146-1537    0.7428    345.0        13324.0   
5196      Santa Ana     CA  92705-5445       NaN     15.0            NaN   
600   West Hartford     CT  06110-1922       NaN    405.0            NaN   

      MN_EARN_WNE_P10  
5028              NaN  
3246              NaN  
3307              NaN  
5763              NaN  
3541              NaN  
1955              NaN  
2152              NaN  
3082              NaN  
5196              NaN  
600               NaN

- Find and display University of Maryland-Baltimore County (UMBC).

In [27]:
# define new data frame which only holds the required institution
df2_filtered = df2[df2["INSTNM"] == "University of Maryland-Baltimore County"]

df2_filtered

UNITID                                   INSTNM       CITY STABBR  \
1478  163268  University of Maryland-Baltimore County  Baltimore     MD   

        ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1478  21250    0.6021  11130.0        11518.0              NaN

- Find and display University of Maryland-College Park.

In [28]:
# define new data frame which only holds the required institution
df2_filtered = df2[df2["INSTNM"] == "University of Maryland-College Park"]

df2_filtered

UNITID                               INSTNM          CITY STABBR    ZIP  \
1479  163286  University of Maryland-College Park  College Park     MD  20742   

      ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1479     0.445  29273.0        10399.0              NaN

- Find and display Loyola University Maryland.

In [29]:
# define new data frame which only holds the required institution
df2_filtered = df2[df2["INSTNM"] == "Loyola University Maryland"]

df2_filtered

UNITID                      INSTNM       CITY STABBR         ZIP  \
1475  163046  Loyola University Maryland  Baltimore     MD  21210-2699   

      ADM_RATE    UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1475    0.7528  3903.0        47560.0              NaN

- Find and display Johns Hopkins University.

In [30]:
# define new data frame which only holds the required institution
df2_filtered = df2[df2["INSTNM"] == "Johns Hopkins University"]

df2_filtered

UNITID                    INSTNM       CITY STABBR         ZIP  \
1473  162928  Johns Hopkins University  Baltimore     MD  21218-2688   

      ADM_RATE    UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1473    0.1254  5663.0        52170.0              NaN

- Calculate the min, max, and average in-state tuition of the above four institutions.

In [31]:
# define new data frame to hold the four required institutions
df2_filtered = df2[df2["INSTNM"].isin(["University of Maryland-Baltimore County", "University of Maryland-College Park", "Loyola University Maryland", "Johns Hopkins University"])]

df2_filtered

UNITID                                   INSTNM          CITY STABBR  \
1473  162928                 Johns Hopkins University     Baltimore     MD   
1475  163046               Loyola University Maryland     Baltimore     MD   
1478  163268  University of Maryland-Baltimore County     Baltimore     MD   
1479  163286      University of Maryland-College Park  College Park     MD   

             ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1473  21218-2688    0.1254   5663.0        52170.0              NaN  
1475  21210-2699    0.7528   3903.0        47560.0              NaN  
1478       21250    0.6021  11130.0        11518.0              NaN  
1479       20742    0.4450  29273.0        10399.0              NaN

In [33]:
# use '.describe' to get descriptive statistics for four institutions
df2_filtered.describe()

UNITID  ADM_RATE          UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10
count       4.000000  4.000000      4.000000       4.000000              0.0
mean   163132.000000  0.481325  12492.250000   30411.750000              NaN
std       174.378898  0.268507  11602.631408   22546.011271              NaN
min    162928.000000  0.125400   3903.000000   10399.000000              NaN
25%    163016.500000  0.365100   5223.000000   11238.250000              NaN
50%    163157.000000  0.523550   8396.500000   29539.000000              NaN
75%    163272.500000  0.639775  15665.750000   48712.500000              NaN
max    163286.000000  0.752800  29273.000000   52170.000000              NaN

df2_filtered.describe shows us the following descriptive statistics for in-state tuition of the above four institutions:
- min - 10399.00
- max - 52170.00
- average - 30411.75

##### 11.  Filter the smaller dataset further to only keep colleges from Maryland, Virgina, Washington DC, Delaware, and Pennsylvinia. Save the filtered dataset to a new file named "Five_States_Colleges.csv" for future analysis. Upload this file to your GitHub along with the notebook.

In [35]:
# create new data frame to hold colleges from the required states
df2_filtered = df2[df2["STABBR"].isin(["MD", "VA", "DC", "DE", "PA"])]

# name and save to csv the data for the 5 states
df2_filtered.to_csv("Five_States_Colleges.csv")

df2_filtered

UNITID                                             INSTNM  \
634     130837  Margaret H Rollins School of Nursing at Beebe ...   
635     130873                          Dawn Career Institute LLC   
636     130907         Delaware Technical Community College-Terry   
637     130934                          Delaware State University   
638     130943                             University of Delaware   
...        ...                                                ...   
7063  45794301   Somerset County Technology Center - Bedford Site   
7084  47503303      Relay Graduate School of Education - Delaware   
7089  47503308  Relay Graduate School of Education - Philadelp...   
7094  47503313  Relay Graduate School of Education - Washingto...   
7103  48305401             Barber School of Pittsburgh - Ambridge   

              CITY STABBR        ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  \
634          Lewes     DE      19958       NaN     59.0            NaN   
635         Newark     DE      19702       NaN    108.0            NaN   
636          Dover     DE      19901       NaN  12408.0         4848.0   
637          Dover     DE      19901    0.4518   3678.0         7868.0   
638         Newark     DE      19716    0.6289  18948.0        13160.0   
...            ...    ...        ...       ...      ...            ...   
7063       Bedford     PA      15522       NaN      NaN            NaN   
7084    Wilmington     DE      19801       NaN      NaN            NaN   
7089  Philadelphia     PA  191224906       NaN      NaN            NaN   
7094    Washington     DC      20002       NaN      NaN            NaN   
7103      Ambridge     PA  150030000       NaN      NaN            NaN   

      MN_EARN_WNE_P10  
634               NaN  
635               NaN  
636               NaN  
637               NaN  
638               NaN  
...               ...  
7063              NaN  
7084              NaN  
7089              NaN  
7094              NaN  
7103              NaN  

[690 rows x 9 columns]